### PR TITLE
Don't use libc's stderr in libdrdecode

### PR DESCRIPTION
**Pull request by khuey**
This fixes a regression introduced in #4567.

### CODE REVIEW

**Comment by khuey** [comment]
Hrm, that fixes the Unix variant platforms but Windows requires something else apparently.

File: core/ir/decodelib.c [hunk 1]
```diff
@@ -128,7 +128,7 @@ get_thread_private_dcontext(void)
 void
 external_error(const char *file, int line, const char *msg)
 {
-    fprintf(stderr, "Usage error: %s (%s, line %d)\n", msg, file, line);
+    print_file(STDERR, "Usage error: %s (%s, line %d)\n", msg, file, line);
```
Comment (derekbruening) [inline core/ir/decodelib.c:131]: Not sure drdecode links utils.c which has print_file.  The `STDERR` should work b/c it turns into either `our_stderr` or `dr_get_stderr_handle` and both are defined up above in this file.  Probably this should be `s/print_file/fprintf/`?

Comment (derekbruening) [reply core/ir/decodelib.c:131]: Never mind, STDERR is the fileno and not the FILE*.  Plus, print_file is defined in this very file.

**Review by derekbruening — APPROVED** [review]

File: core/ir/decodelib.c [hunk 2]
```diff
@@ -128,7 +128,7 @@ get_thread_private_dcontext(void)
 void
```
Comment (derekbruening) [inline core/ir/decodelib.c:128]: The commit title and description are not following the conventions in multiple respects:

+ The title should start with the issue number
+ The description should describe what is being fixed (yes, duplicating the title)
+ There should be a Fixes line at the bottom

Please see https://github.com/DynamoRIO/dynamorio/wiki/Code-Reviews#commit-messages

**Review by derekbruening — CHANGES_REQUESTED** [review]

**Comment by khuey** [comment]
Thanks for approving this I guess but I still don't know how to make it build on Windows :)

**Comment by derekbruening** [comment]
Pasting the error would make it easier so we don't have to go click on the logs too.  Needs get_stderr_handle, etc. equivalent to the dr_ ones right?

**Comment by khuey** [comment]
```
 ====> FAILURE in debug-internal-32 <====
debug-internal-32: **** 2 build errors ****
	drdecode.lib(decodelib.c.obj) : error LNK2019: unresolved external symbol _get_stderr_handle referenced in function _external_error
	clients\bin32\drdisas.exe : fatal error LNK1120: 1 unresolved externals
```

**Comment by khuey** [comment]
I guess I could s/STDERR/dr_get_stderr_file()/ perhaps?

**Comment by derekbruening** [comment]
globals_shared.h has `define STDERR (dr_get_stderr_file())` which is in decodelib.c, but globals.h (internal) has `#    define STDERR (file_t) get_stderr_handle()` which is not in decodelib.c -- maybe it needs both or an alias or whatnot?

**Comment by derekbruening** [comment]
I would add get_stderr_handle which calls dr_get_stderr_handle or sthg

**Comment by derekbruening** [comment]
> I guess I could s/STDERR/dr_get_stderr_file()/ perhaps?

Also seems good -- I guess the internal define is being used for decodelib itself and so there is no other reason for the non dr_ versions.  External users of the lib are going to have the external define and thus need the dr_ version.

**Comment by khuey** [comment]
> 
> 
> > I guess I could s/STDERR/dr_get_stderr_file()/ perhaps?
> 
> Also seems good -- I guess the internal define is being used for decodelib itself and so there is no other reason for the non dr_ versions. External users of the lib are going to have the external define and thus need the dr_ version.

Doing this would require defining dr_get_stderr_file for the Unix platforms as its also windows only.

**Comment by khuey** [comment]
Alright how about that?

**Review by derekbruening — APPROVED** [review]
Works for me.  Don't forget the commit title + description conventions.

**Pull request merged**